### PR TITLE
DNM: network driver

### PR DESCRIPTION
**Pull request by mmoskal**
CC @jamesadevine

### CODE REVIEW

File: libs/network/network.cpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    JDDevice *getDevice() { return &device; }
+
+    ~JDProxyDriver() { decrRC(methods); }
+};
+
+//%
+JDProxyDriver *addNetworkDriver(int driverClass, RefCollection *methods) {
+    static int deviceId = 3030;
+    return new JDProxyDriver(JDDevice(0, 0, 0, driverClass), methods, ++deviceId);
```
Comment (jamesadevine) [inline libs/network/network.cpp]: You'll need to pass flags which tells the logic driver what mode to run the driver in. I've got a commit that has an enum instead of a #define for the flags.

Comment (mmoskal) [reply libs/network/network.cpp]: Sure thing. I'm assuming you're handling the serial number?

Comment (jamesadevine) [reply libs/network/network.cpp]: Will do.

File: libs/network/network.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+namespace network {
+    //% shim=pxt::programHash
+    export function programHash(): int32 { return 0 }
+
+    //% shim=network::addNetworkDriver
+    function addNetworkDriver(deviceClass: int32, methods: ((p:Buffer) => void)[]): NetworkDriverStatus { 
+        return null
+    }
+
+    export function addDriver(deviceClass: int32, n: NetworkDriver) {
```
Comment (pelikhan) [inline libs/network/network.ts]: deviceClass not used?

File: libs/network/pxt.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
+    "name": "network",
```
Comment (pelikhan) [inline libs/network/pxt.json]: should we have a more unique name? network could be confusing.

File: libs/base/core.cpp [hunk 4]
```diff
@@ -482,11 +482,11 @@ TValue ptrneqq(TValue a, TValue b) {
 }
 } // namespace langsupp
 
+
 #define NUMOP(op) return fromDouble(toDouble(a) op toDouble(b));
 #define BITOP(op) return fromInt(toInt(a) op toInt(b));
 namespace numops {
 
-//%
```
Comment (pelikhan) [inline libs/base/core.cpp]: @mmoskal  are these changes needed or just a merge fluke?

**Comment by jamesadevine** [comment]
👍
